crates/bevy_reflect/src/type_registry.rs

```diff
@@ -92,24 +99,53 @@ impl TypeRegistry {
         registry
     }
 
-    /// Registers the type `T`, adding reflect data as specified in the [`Reflect`] derive:
+    /// Attempts to register the type `T` if it has not yet been registered already.
+    ///
+    /// If the registration for type `T` already exists, it will not be registered again.
+    /// To register the type, overwriting any existing registration, use [register](Self::register) instead.
+    ///
+    /// Additionally, this will add the reflect [data](TypeData) as specified in the [`Reflect`] derive:
     /// ```rust,ignore
     /// #[derive(Reflect)]
     /// #[reflect(Component, Serialize, Deserialize)] // will register ReflectComponent, ReflectSerialize, ReflectDeserialize
+    /// struct Foo;
     /// ```
     pub fn register<T>(&mut self)
     where
         T: GetTypeRegistration,
     {
-        self.add_registration(T::get_type_registration());
+        if !self.add_registration(T::get_type_registration()) {
```

Comment:
This could avoid computing `T::get_type_registration` is `T` is already registered (assuming `T: GetTypeRegistration always returns a `TypeRegistration` for `T` and not for other types)